api/v1/nodetemplate_types.go

```diff
@@ -0,0 +1,68 @@
+package v1
+
+import (
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+type NodeConfigurationSpec struct {
+	Taints []corev1.Taint `json:"taints,omitempty"`
+}
+
+type NodeConfigurationMetadata struct {
+	Annotations map[string]string `json:"annotations,omitempty"`
+	Labels      map[string]string `json:"labels,omitempty"`
+}
+
+type NodeConfiguration struct {
+	Metadata NodeConfigurationMetadata `json:"metadata,omitempty"`
+	Spec     NodeConfigurationSpec     `json:"spec,omitempty"`
+}
+
+type NodeTemplateSpec struct {
+	// +kubebuilder:default: false
+	DryRun bool `json:"dryRun,omitempty"`
+	// +kubebuilder:default: 1
+	Priority int                   `json:"priority"`
+	Selector *metav1.LabelSelector `json:"selector"`
+	// +kubebuilder:validation:Minimum=1
+	Nodes    int               `json:"nodes"`
+	Template NodeConfiguration `json:"template,omitempty"`
+}
+
+type ReconcileInfo struct {
+	Generation int64 `json:"generation,omitempty"`
+}
+
+type NodeTemplateStatus struct {
+	CurrentNodes  int           `json:"currentNodes"`
+	Sufficient    bool          `json:"sufficient"`
+	ReconcileInfo ReconcileInfo `json:"reconcileInfo"`
+}
+
+// +kubebuilder:object:root=true
+// +kubebuilder:subresource:status
+// +kubebuilder:resource:scope=Cluster
+// +kubebuilder:printcolumn:name="current",type="integer",JSONPath=`.status.currentNodes`,description="Current Nodes"
+// +kubebuilder:printcolumn:name="expected",type="integer",JSONPath=`.spec.nodes`,description="Expected Nodes"
```

Comment:
This value is a request. So `desired` is more suitable, IMO.
```suggestion
// +kubebuilder:printcolumn:name="desired",type="integer",JSONPath=`.spec.nodes`,description="Desired Nodes"
```

FYI, in standard k8s resources, `desired` is used for values such as this.
```console
$ kubectl explain daemonset.status.desiredNumberScheduled
GROUP:      apps
KIND:       DaemonSet
VERSION:    v1

FIELD: desiredNumberScheduled <integer>


DESCRIPTION:
    The total number of nodes that should be running the daemon pod (including
    nodes correctly running the daemon pod). More info:
    https://kubernetes.io/docs/concepts/workloads/controllers/daemonset/
    

$ kubectl explain statefulset.spec.replicas
GROUP:      apps
KIND:       StatefulSet
VERSION:    v1

FIELD: replicas <integer>


DESCRIPTION:
    replicas is the desired number of replicas of the given Template. These are
    replicas in the sense that they are instantiations of the same Template, but
    individual replicas also have a consistent identity. If unspecified,
    defaults to 1.
    

```